src/map.rs

```diff
@@ -1218,6 +1247,44 @@ where
         None
     }
 
+    /// Retains only the elements specified by the predicate.
+    ///
+    /// In other words, remove all pairs (k, v) such that f(&k,&v) returns false.
+    ///
+    /// This method relies on [`HashMap::replace_node`] and do not force entries deletion if the
```

Comment:
Since `replace_node` is not public, I don't think it should be mentioned in the docs. How about:
```rust
/// If `f` returns `false` for a given key/value pair, but the value for that pair is concurrently modified before the removal takes place, the entry will not be removed. If you want the removal to happen even in the case of concurrent modification, use [`HashMap::retain_force`].